app/javascript/packs/admin.tsx

```diff
@@ -211,49 +239,52 @@ const onChangeRegistrationMode = (target) => {
     });
 };
 
-const convertUTCDateTimeToLocal = (value) => {
+const convertUTCDateTimeToLocal = (value: string) => {
   const date = new Date(value + 'Z');
-  const twoChars = (x) => x.toString().padStart(2, '0');
+  const twoChars = (x: number) => x.toString().padStart(2, '0');
   return `${date.getFullYear()}-${twoChars(date.getMonth() + 1)}-${twoChars(date.getDate())}T${twoChars(date.getHours())}:${twoChars(date.getMinutes())}`;
 };
 
-const convertLocalDatetimeToUTC = (value) => {
-  const re = /^([0-9]{4,})-([0-9]{2})-([0-9]{2})T([0-9]{2}):([0-9]{2})/;
-  const match = re.exec(value);
-  const date = new Date(match[1], match[2] - 1, match[3], match[4], match[5]);
+function convertLocalDatetimeToUTC(value: string) {
+  const date = new Date(value);
   const fullISO8601 = date.toISOString();
   return fullISO8601.slice(0, fullISO8601.indexOf('T') + 6);
-};
+}
```

Comment:
I remember I came up with the parsing regexp in https://github.com/mastodon/mastodon/pull/18321/commits/4cc0fcb11994392ebde1c079b09b383093213b66 to ensure it was correctly parsed with the correct local timezone. I remember there being caveats but I unfortunately can't remember which.

According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format, this should be the one format that is correctly handled.

I'm not sure whether there really is an issue, but I suppose this is worth checking across browsers.

Reply:
I tested a few modern browsers and they seem to behave as expected. I also can't find any reference suggesting that older browsers would do otherwise. So this seems fine to me.